concordium-std/src/test_infrastructure.rs

```diff
@@ -1460,9 +1478,46 @@ impl<State: Serial + DeserialWithState<TestStateApi> + StateClone<TestStateApi>>
         if amount.micro_ccd > 0 {
             *self.contract_balance.borrow_mut() -= amount;
         }
+        // Update the receiver query balance.
+        if let Some(balance) = self.query_contract_balances.borrow_mut().get_mut(to) {
+            *balance += amount;
+        }
+
         Ok(res)
     }
 
+    fn account_balance(&self, address: AccountAddress) -> QueryAccountBalanceResult {
+        if self.missing_accounts.contains(&address) {
+            Err(QueryAccountBalanceError)
+        } else if let Some(balances) = self.query_account_balances.borrow().get(&address) {
+            Ok(*balances)
+        } else {
+            fail!("No account balance for {:?} has been set up.", address)
```

Comment:
Why not use the display implementation? Is it not available here?

Reply:
It is not available here.
We need base58check which is only available when `derive-serde` is enabled in `concordium-contracts-common`